src/containers/TriggerTemplate/TriggerTemplate.test.js

```diff
@@ -0,0 +1,97 @@
+/*
+Copyright 2019 The Tekton Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React from 'react';
+import { waitForElement } from 'react-testing-library';
+import { Provider } from 'react-redux';
+import thunk from 'redux-thunk';
+import configureStore from 'redux-mock-store';
+import { createIntl } from 'react-intl';
+import { TriggerTemplateContainer } from './TriggerTemplate';
+import { renderWithRouter } from '../../utils/test';
+
+beforeEach(jest.resetAllMocks);
+
+const intl = createIntl({
+  locale: 'en',
+  defaultLocale: 'en'
+});
+
+it('TriggerTemplateContainer renders', async () => {
+  const triggerTemplateName = 'bar';
+  const match = {
+    params: {
+      triggerTemplateName
+    }
+  };
+  const middleware = [thunk];
+  const mockStore = configureStore(middleware);
+  const testStore = mockStore({
+    namespaces: {
+      selected: 'default'
+    },
+    triggerTemplates: {
+      byId: {},
+      byNamespace: { default: {} },
+      errorMessage: null,
+      isFetching: false
+    }
+  });
+
+  const { getByText } = renderWithRouter(
+    <Provider store={testStore}>
+      <TriggerTemplateContainer
+        match={match}
+        fetchTriggerTemplate={() => Promise.resolve()}
+        error={null}
+        loading={false}
+      />
+    </Provider>
+  );
+  await waitForElement(() => getByText(`TriggerTemplate not available`));
+});
+
+it('TriggerTemplateContainer handles error state', async () => {
+  const match = {
+    params: {
+      triggerTemplateName: 'foo'
+    }
+  };
+
+  const middleware = [thunk];
+  const mockStore = configureStore(middleware);
+
+  const testStore = mockStore({
+    namespaces: {
+      selected: 'default'
+    },
+    triggerTemplates: {
+      byId: {},
+      byNamespace: { default: {} },
+      errorMessage: 'Error',
+      isFetching: false
+    }
+  });
+
+  const { getByText } = renderWithRouter(
+    <Provider store={testStore}>
+      <TriggerTemplateContainer
+        intl={intl}
+        match={match}
+        error="Error"
+        fetchTriggerTemplate={() => Promise.resolve()}
+      />
+    </Provider>
+  );
+  await waitForElement(() => getByText('Error loading TriggerTemplate'));
+});
```

Comment:
don't see any test cases for accordion clicks and rows rendered properly.

Reply:
Adding

Reply:
Ok added lots of tests to ensure the data is good, I didn't add any on click business because the elements are already on the page anyway 🤔